### PR TITLE
fix: download linked objects in a WFS using "resolvedepth"

### PR DESCRIPTION
**Pull request by stempler**
Using "resolvedepth" when trying to download data via WFS is now downloading all the linked objects.

ING-4128

### CODE REVIEW

File: io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/DuplicateIDsFilterIterator.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	/**
+	 * @return Instance
+	 * @see java.util.Iterator#next()
+	 */
+	@Override
+	public Instance next() {
+		hasNext();
+		if (nextInstance == null) {
+			for (String string : uniqueIDInstances) {
+				System.out.println(string);
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/DuplicateIDsFilterIterator.java:142]: Should be removed

**Review by emanuelaepure10 — APPROVED** [review]
LGTM

File: io/plugins/eu.esdihumboldt.hale.io.gml/test/eu/esdihumboldt/hale/io/gml/reader/internal/GmlInstanceCollectionTest.java [hunk 2]
```diff
@@ -262,6 +272,58 @@ public void testLoadShiporder() throws Exception {
...
+	 * @throws URISyntaxException
+	 */
+	private int extractedUniqueInstances(String pathToXML)
+			throws IOException, IOProviderConfigurationException, URISyntaxException {
+		GmlInstanceCollection instances = loadInstances(
+				getClass().getResource("/data/filter_doubles/filter.xsd").toURI(),
+				getClass().getResource(pathToXML).toURI(), false);
+
+		FullInstanceIteratorSupport fullInstance = new FullInstanceIteratorSupport(
+				instances.iterator());
```
Comment (florianesser) [inline io/plugins/eu.esdihumboldt.hale.io.gml/test/eu/esdihumboldt/hale/io/gml/reader/internal/GmlInstanceCollectionTest.java:311]: see [my comment here](https://github.com/halestudio/hale/pull/1180/files#r1725319325)

Comment (stempler) [reply io/plugins/eu.esdihumboldt.hale.io.gml/test/eu/esdihumboldt/hale/io/gml/reader/internal/GmlInstanceCollectionTest.java]: Removed the use of FullInstanceIteratorSupport here.

**Pull request merged**